src/scripts/prepareForExtensionPublish.ts

```diff
@@ -0,0 +1,59 @@
+import * as semver from "semver";
+import { exec } from "child_process";
+import { promisify } from "util";
+import { writeFile } from "fs/promises";
+
+const execAsync = promisify(exec);
+
+/**
+ * Prepares the directory for extension publication. Does the following:
+ *
+ * 1. Changes the package version so that the patch number is the number of
+ *    commits on the current branch
+ * 2. Writes a file called `build-info.json` for provenance
+ */
+async function main() {
+  const { major, minor } = semver.parse(process.env.npm_package_version)!;
+
+  const commitCount = await runCommand("git rev-list --count head");
```

Comment:
Should this be spelt `HEAD`?

Reply:
Works lowercase too, but agreed `HEAD` is more canonical